template/tsconfig.test.json

```diff
@@ -0,0 +1,4 @@
+{
+  "extends": "tsconfig.json",
+  "include": ["src/**/*.tsx", "src/**/*.ts"]
+}
```

Comment:
This is a bit awkward but I figured the best way to go about the issue. 

To ensure the tests directory isn't bundled into our output, I exclude everything but `components` from the main `tsconfig.json`. However, my IDE (Webstorm), and I'm sure others, will raise errors in the non-`components` files as now there are `.ts` files that are not covered by a `tsconfig.json`. This is reasonable, the `--jsx` flag is needed. However, this PR uses `jest-preset-preact` to do our ts->js transformations, and it just uses Babel. It doesn't care about our config one bit. 

So this config is essentially just here to silence any warnings users might have. They can tell their IDEs to use this config, which covers all files, while the main one will just cover the files we actually want to output.

There might be a better way to go about this, not 100% sure.